databricks/sdk/environments.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from .azure import ARM_DATABRICKS_RESOURCE_ID, ENVIRONMENTS, AzureEnvironment
+
+
+class Cloud(Enum):
+    AWS = "AWS"
+    AZURE = "AZURE"
+    GCP = "GCP"
+
+
+@dataclass
+class DatabricksEnvironment:
+    cloud: Cloud
+    dns_zone: str
+    azure_application_id: Optional[str] = None
+    azure_environment: Optional[AzureEnvironment] = None
+
+    def deployment_url(self, name: str) -> str:
+        return f"https://{name}.{self.dns_zone}"
+
+    @property
+    def azure_service_management_endpoint(self) -> Optional[str]:
+        if self.azure_environment is None:
+            return None
+        return self.azure_environment.service_management_endpoint
+
+    @property
+    def azure_resource_manager_endpoint(self) -> Optional[str]:
+        if self.azure_environment is None:
+            return None
+        return self.azure_environment.resource_manager_endpoint
+
+    @property
+    def azure_active_directory_endpoint(self) -> Optional[str]:
+        if self.azure_environment is None:
+            return None
+        return self.azure_environment.active_directory_endpoint
+
+
+DEFAULT_ENVIRONMENT = DatabricksEnvironment(Cloud.AWS, ".cloud.databricks.com")
+
+ALL_ENVS = [
+    DatabricksEnvironment(Cloud.AWS, ".dev.databricks.com"),
+    DatabricksEnvironment(Cloud.AWS, ".staging.cloud.databricks.com"),
+    DatabricksEnvironment(Cloud.AWS, ".cloud.databricks.us"), DEFAULT_ENVIRONMENT,
```

Comment:
Nit, put DEFAULT_ENVIRONMENT in a new line (maybe in the first line).

Reply:
This was done by our formatter, so I think I wil leave this as is.